src/main/java/ch/njol/skript/expressions/ExprTabCompletionArgs.java

```diff
@@ -0,0 +1,103 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.event.server.TabCompleteEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Tab Completion Arguments")
+@Description("Get the arguments used within a tab complete event")
+@Examples({"on tab complete of \"test\":",
+	"\ttab argument 1 is \"bar\"",
+	"\tset tab completions for position 2 to \"foo1\",\"foo2\", and \"foo3\""})
+@Events("tab complete")
+@Since("INSERT VERSION")
+public class ExprTabCompletionArgs extends SimpleExpression<String> {
```

Comment:
Currently this is returning only strings.

I think if the command input is aimed at a Skript command the arguments string should be parsed using the command parser and use the regular arguments syntax (e.g. arg-text, arg-number) since that is what users are used to.